### PR TITLE
Adjusting code to use protobuf files

### PR DESCRIPTION
**Pull request by moniercoding**

### CODE REVIEW

File: lib/src/zenohRpcClient.cpp [hunk 1]
```diff
@@ -127,103 +125,84 @@ std::future<UPayload> ZenohRpcClient::invokeMethod(const UUri &uri,
...
+        spdlog::error("Wrong message type = {}", static_cast<int>(attributes.type()));
         return std::move(future);
     }
 
-    auto uriHash = std::hash<std::string>{}(LongUriSerializer::serialize(uri));
-
-    auto header = MessageBuilder::buildHeader(attributes);
-    if (header.empty()) {
-        spdlog::error("Failed to build header");
+    // Serialize UAttributes
```
Comment (MelamudMichael) [inline lib/src/zenohRpcClient.cpp:133]: move serialization just before z_get

File: lib/src/zenohRpcClient.cpp [hunk 2]
```diff
@@ -127,103 +125,84 @@ std::future<UPayload> ZenohRpcClient::invokeMethod(const UUri &uri,
...
-            spdlog::error("No attachment found in the reply");
-            break;
-        }       
-
-        z_bytes_t index = z_attachment_get(sample.attachment, z_bytes_new("header"));
 
-        if (!z_check(index)) {
-            spdlog::error("Header not found in the attachment");
-            break;
+        uprotocol::v1::UMessage message;
```
Comment (MelamudMichael) [inline lib/src/zenohRpcClient.cpp:185]: why umessage ? it should be uattributes..

File: lib/src/zenohRpcClient.cpp [hunk 3]
```diff
@@ -127,103 +125,84 @@ std::future<UPayload> ZenohRpcClient::invokeMethod(const UUri &uri,
...
+            spdlog::error("ParseFromArray failure");
+            continue;
         }
 
-        auto allTlv = MessageParser::getAllTlv(reinterpret_cast<const uint8_t*>(index.start), index.len);
-        if (!allTlv.has_value()) {
-            spdlog::error("MessageParser::getAllTlv failure");
-            break;
-        }
+        const auto& payloadData = message.payload();
```
Comment (MelamudMichael) [inline lib/src/zenohRpcClient.cpp:191]: upayload should receive the data from sample.payload

File: lib/src/zenohUTransport.cpp [hunk 4]
```diff
@@ -508,138 +483,145 @@ UStatus ZenohUTransport::unregisterListener(const UUri &uri,
...
+       spdlog::error("Serialized attributes not found in the attachment");
+       return;
     }
 
-    // TLV extraction
-    auto allTlv = MessageParser::getAllTlv(index.start, index.len);
-    if (!allTlv.has_value()) {
-        spdlog::error("MessageParser::getAllTlv failure");
-        return;
+    uprotocol::v1::UAttributes attributes;
```
Comment (MelamudMichael) [inline lib/src/zenohUTransport.cpp:505]: use this as an example

File: lib/src/zenohUTransport.cpp [hunk 5]
```diff
@@ -508,138 +483,145 @@ UStatus ZenohUTransport::unregisterListener(const UUri &uri,
...
-        }
+            break;
+        default:
+            return UCode::UNAVAILABLE;
     }
 
     return UCode::OK;
 }
+// UCode ZenohUTransport::mapEncoding(const USerializationHint &encodingIn, 
+//                                    z_encoding_t &encodingOut) noexcept {
```
Comment (MelamudMichael) [inline lib/src/zenohUTransport.cpp:595]: delete this function

**Review by MelamudMichael — CHANGES_REQUESTED** [review]

**Comment by PLeVasseur** [comment]
Hey @moniercoding :wave: -- could you rebase this PR? Is it up for another round of review?

**Comment by stevenhartley** [comment]
Pr has gone stale, closing

**Pull request closed without merging**